src/hooks/data/useData.ts

```diff
@@ -0,0 +1,120 @@
+import http from 'api';
+import * as Type from 'api/types';
+import { useQuery } from 'react-query';
+
+interface Response {
+  message: string;
+}
+
+function getMyPage() {
+  return http.get<Type.MyInfo>('/users/mypage');
+}
+
+function editMyPage(data: Type.MyInfo) {
+  return http.put<Response, Type.MyInfo>('/users/mypage', data)
+}
```

Comment:
```suggestion
function editMyPage(data: Type.MyInfo) {
  return http.put<Response>('/users/mypage', data);
}
```

요렇게만 해도 `RequestBody` 제네릭타입은 추론될 것 같아유!

Reply:
혹여나 에러나면 api함수들의 제네릭타입에
```ts
<ResponseBody, RequestBody = unknown>
```
처럼 기본 타입을 넣어주면 될 것 같슴다